onefuse/resource_fuse_naming.go

```diff
@@ -87,21 +72,50 @@ func bindResource(d *schema.ResourceData, cn CustomName) error {
 	return nil
 }
 
+func resourceCustomNameCreate(d *schema.ResourceData, m interface{}) error {
+	log.Println("onefuse.resourceCustomNameCreate")
+
+	config := m.(Config)
+
+	dnsSuffix := d.Get("dns_suffix").(string)
+	namingPolicyID := d.Get("naming_policy_id").(string)
+	workspaceID := d.Get("workspace_id").(string)
+	templateProperties := d.Get("template_properties").(map[string]interface{})
+
+	cn, err := config.NewOneFuseApiClient().GenerateCustomName(dnsSuffix, namingPolicyID, workspaceID, templateProperties)
+	if err != nil {
+		return err
+	}
+
+	return bindCustomNamingResource(d, cn)
+}
+
 func resourceCustomNameRead(d *schema.ResourceData, m interface{}) error {
+	log.Println("onefuse.resourceCustomNameRead")
+
 	config := m.(Config)
+
 	id := d.Get("custom_name_id").(int)
+
 	customName, err := config.NewOneFuseApiClient().GetCustomName(id)
-	bindResource(d, customName)
-	return err
+	if err != nil {
+		return err
+	}
+
+	return bindCustomNamingResource(d, customName)
 }
 
 func resourceCustomNameUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Println("onefuse.resourceCustomNameUpdate")
 	return nil
 }
 
 func resourceCustomNameDelete(d *schema.ResourceData, m interface{}) error {
+	log.Println("onefuse.resourceCustomNameDelete")
+
 	config := m.(Config)
+
 	id := d.Get("custom_name_id").(int)
-	config.NewOneFuseApiClient().DeleteCustomName(id)
-	return nil
+
+	return config.NewOneFuseApiClient().DeleteCustomName(id)
```

Comment:
TODO: We want to catch errors like "not authorized" or "server error" but we want to ignore "404 not found" errors because that implies the object was successfully deleted.

Reply:
@pop - nitpick, IMO, successful deletion should result in a 204 no content. Are there instances where that's not the case in the OneFuse API?

Reply:
That is the case. `DeleteName` just returns an error. Our SDK returns an error if the API responds with a 400 or 500-class error.

Do you think we should be doing more checking here?